djconnectwise/sync.py

```diff
@@ -126,7 +126,11 @@ def _assign_relation(self, instance, json_data,
             self._assign_null_relation(instance, model_field)
             return
 
-        uid = relation_json['id']
+        if isinstance(relation_json, int):
+            uid = relation_json
+        else:
+            uid = relation_json['id']
```

Comment:
The reason for this is that usually CW refers to FK relationships by providing a `dict` (well gets converted to a dict when converted to python) in a form like
```python
"project": {
     "id": 3,
     "name": Project name,
     etc...
}
```
but in this case CW just gives us the project id as `projectId`: 3. 

But I do agree that this solution is probably asking for trouble. Probably better to override the `_assign_relation` method or maybe even tweak the json to be in the form mentioned above. 

Any thoughts on a better solution?

Reply:
Ah yes, this is a pattern we already have for schedule entries and service notes, they have the same problem but with `objectId` and `ticketId` respectively. Schedule entries are a little special because they have to deal with opportunities and tickets, so look to ServiceNoteSynchronizer for an example of what needs to be done.